python/lsst/dax/imgserv/api_soda.py

```diff
@@ -124,45 +143,272 @@ def img_capabilities():
+            image.writeFits(fp.name)
+            resp = send_file(fp.name,
+                             mimetype="image/fits",
+                             as_attachment=True,
+                             attachment_filename=os.path.basename(fp.name))
+            return resp
     else:
         return _image_not_found()
 
 
 @image_soda.route("/async", methods=["GET", "POST"])
 def img_async():
+    """ Get the /async service endpoint. """
     if not request.args:
         # no parameters present, return service status
         soda_url = url_for('api_image_soda.img_async', _external=True)
         return _service_response(soda_url)
-    xml = current_app.soda.do_async(_getparams())
-    resp = make_response(xml)
-    resp.headers["Content_Type"] = "text/xml"
-    return resp
+    job_id = current_app.soda.do_async(_getparams())
+    return redirect(url_for('api_image_soda.img_async_jobs_job',
+                            job_id=job_id,
+                            _external=True))
+
+
+@image_soda.route("/async-jobs", methods=["GET"])
+def img_async_jobs():
+    """ Get all the jobs of the user in the system. """
+    if session.get("user"):
+        xml = current_app.soda.get_jobs(_getparams())
+        resp = make_response(xml)
+        resp.headers["Content-Type"] = "text/xml"
+        return resp
+    else:
+        raise Exception("Error: user not specified")
+
+
+@image_soda.route("/async-jobs/<job_id>", methods=["GET"])
+def img_async_jobs_job(job_id: str):
+    """ Get the job info, else path to the result if ready.
+
+    Parameters
+    ----------
+    job_id: `str`
+
+    Returns
+    -------
+    xml: `str`
+        the job description.
+    """
+    ar = app_celery.AsyncResult(job_id)
+    if ar.state == "SUCCESS":
+        return redirect(url_for('api_image_soda.img_async_job_results',
+                                job_id=job_id,
+                                _external=True))
+
+    else:
+        job_result_url = url_for('api_image_soda.img_async_job_results',
+                                 job_id = job_id,
+                                 _external=True)
+        return _uws_job_response(job_result_url)
+
+
+@image_soda.route("/async-jobs/<job_id>/phase", methods=["GET"])
+def img_async_job_phase(job_id: str):
+    """ Get the phase info for the job.
+
+    Parameters
+    ----------
+    job_id : `str`
+
+    Returns
+    -------
+    xml: `str`
+        phase for the job.
+    """
+    ar = app_celery.AsyncResult(job_id)
+    phase = map_phase_from_state[ar.state]
+    return _uws_job_response_plain("PHASE="+phase)
+
+
+@image_soda.route("/async-jobs/<job_id>/executionduration", methods=["GET"])
+def img_async_job_duration(job_id: str):
+    """ Get the duration in number of seconds for the job.
+
+    Parameters
+    ----------
+    job_id : `str`
+
+    Returns
+    -------
+    xml: `str`
+        the job duration info.
+    """
+    return _uws_job_response_plain("INFO=NOT IMPLEMENTED")
```

Comment:
You should probably throw here, or at least mark it as something that won't return a 200 here for a not implemented.  Ditto for other places using this pattern.

Reply:
yep